### PR TITLE
ZCS-9595:- Status of expanded/collapsed of a folder is not saved when zimbraFeatureOptionsEnabled is FALSE

### PR DESCRIPTION
**Pull request by madhavdhuppe**
Removed OPTIONS_ENABLED check when use is doing implicit settings.

### CODE REVIEW

**Review by BhavinBathani — APPROVED** [review]
LGTM

File: WebRoot/js/zimbraMail/share/model/ZmSettings.js [hunk 1]
```diff
@@ -1203,9 +1203,6 @@ ZmSettings.prototype._showConfirmDialog = function(text, callback, style) {
 
 ZmSettings.prototype._implicitChangeListener =
 function(ev) {
-	if (!appCtxt.get(ZmSetting.OPTIONS_ENABLED)) {
```
Comment (madhavdhuppe) [inline WebRoot/js/zimbraMail/share/model/ZmSettings.js:1206]: Removing this check to apply implicit setting change listener.

is that fine @silentsakky @BhavinBathani ?

Comment (silentsakky) [reply WebRoot/js/zimbraMail/share/model/ZmSettings.js:1206]: this function does more then just setting expand/collapse status of folder, looks like we are breaking that functionality

Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/share/model/ZmSettings.js:1206]: ok then will add check to only allow expand/collapse status of folder

File: WebRoot/js/zimbraMail/share/model/ZmSettings.js [hunk 2]
```diff
@@ -1203,11 +1203,12 @@ ZmSettings.prototype._showConfirmDialog = function(text, callback, style) {
...
+	var id = ev.source.id;
+
+	if (!appCtxt.get(ZmSetting.OPTIONS_ENABLED) && ZmSetting.FOLDERS_EXPANDED !== id) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/share/model/ZmSettings.js:1208]: better to move single if condition to line no 1219
after we have done processing for expand/collapse folder save

Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/share/model/ZmSettings.js:1208]: ok

**Review by silentsakky — APPROVED** [review]
Please update commit message as well

File: WebRoot/js/zimbraMail/share/model/ZmMetaData.js [hunk 3]
```diff
@@ -105,6 +105,14 @@ function(section, data, batchCommand, callback, errorCallback, sensitive) {
 		if (AjxUtil.isObject(d)) {
 			d = ZmSetting.serialize(d, ZmSetting.D_HASH);
 		}
+
+		// ToDo:- Remove this code after ZCS-9866 is closed.
+		if (i === "zimbraPrefFoldersExpanded" && (d === "" || d == null)) {
+			// Adding hardcoded value 3:true to always send non empty value of zimbraPrefFoldersExpanded
+			// bacause as of now backend doesn't support empty values of key.
+			d = '3:true';
```
Comment (dajay0) [inline WebRoot/js/zimbraMail/share/model/ZmMetaData.js:113]: Add to the comment above: 3 is drafts folder. does not have subfolders. hence this is a noop

Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/share/model/ZmMetaData.js:113]: sure

**Review by BhavinBathani — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

File: WebRoot/js/zimbraMail/share/model/ZmMetaData.js [hunk 4]
```diff
@@ -105,8 +105,13 @@ function(section, data, batchCommand, callback, errorCallback, sensitive) {
 		if (AjxUtil.isObject(d)) {
 			d = ZmSetting.serialize(d, ZmSetting.D_HASH);
 		}
-		var a = soapDoc.set("a", d, metaNode);
-		a.setAttribute("n", i);
+
+		// Don't set metadata key values if value is null or empty.
+		// SetMailboxMetadataRequest doesn't support empty value of key.
```
[the remark this answers is not in the document]
Comment (madhavdhuppe) [reply WebRoot/js/zimbraMail/share/model/ZmMetaData.js:110]: updated

**Review by silentsakky — APPROVED** [review]

**Pull request merged**